c7n/policy.py

```diff
@@ -893,8 +902,9 @@ def execution_mode(self):
         return self.data.get('mode', {'type': 'pull'})['type']
 
     def get_execution_mode(self):
-        exec_mode = execution[self.execution_mode]
-        if exec_mode is None:
+        try:
+            exec_mode = execution[self.execution_mode]
+        except KeyError:
```

Comment:
given the feedback we should probably not catch this and instead fix the test of invalid provider to expect such an error.